arch/common/isr_tables.c compilation fails with CONFIG_NUM_IRQS=0

**Describe the bug**
The `isr_tables.c` file fails to compile with CONFIG_NUM_IRQS=0. This was found when reviewing #27394 along with @ioannisg.

**To Reproduce**
Steps to reproduce the behavior:
1. Change a SoC definition to set `CONFIG_NUM_IRQS=0`
2. cd samples/hello_world
3. mkdir build; cd build
4. cmake -DBOARD=some\_arm\_board\_with\_given\_soc -DCONFIG_UART_INTERRUPT_DRIVEN=n
5. make
6. See error:

```
Scanning dependencies of target arch__common
[ 56%] Building C object zephyr/arch/common/CMakeFiles/arch__common.dir/isr_tables.c.obj
/home/brix/Projects/zephyrproject/zephyr/arch/common/isr_tables.c:54:3: error: array index in initializer exceeds array bounds
   54 |  [0 ...(IRQ_TABLE_SIZE - 1)] = (uint32_t)&IRQ_VECTOR_TABLE_DEFAULT_ISR,
      |   ^
/home/brix/Projects/zephyrproject/zephyr/arch/common/isr_tables.c:54:3: note: (near initialization for '_irq_vector_table')
/home/brix/Projects/zephyrproject/zephyr/arch/common/isr_tables.c:54:32: warning: excess elements in array initializer
   54 |  [0 ...(IRQ_TABLE_SIZE - 1)] = (uint32_t)&IRQ_VECTOR_TABLE_DEFAULT_ISR,
      |                                ^
/home/brix/Projects/zephyrproject/zephyr/arch/common/isr_tables.c:54:32: note: (near initialization for '_irq_vector_table')
/home/brix/Projects/zephyrproject/zephyr/arch/common/isr_tables.c:63:3: error: array index in initializer exceeds array bounds
   63 |  [0 ...(IRQ_TABLE_SIZE - 1)] = {(void *)0x42, (void *)&z_irq_spurious},
      |   ^
/home/brix/Projects/zephyrproject/zephyr/arch/common/isr_tables.c:63:3: note: (near initialization for '_sw_isr_table')
/home/brix/Projects/zephyrproject/zephyr/arch/common/isr_tables.c:63:32: warning: excess elements in array initializer
   63 |  [0 ...(IRQ_TABLE_SIZE - 1)] = {(void *)0x42, (void *)&z_irq_spurious},
      |                                ^
/home/brix/Projects/zephyrproject/zephyr/arch/common/isr_tables.c:63:32: note: (near initialization for '_sw_isr_table')
zephyr/arch/common/CMakeFiles/arch__common.dir/build.make:82: recipe for target 'zephyr/arch/common/CMakeFiles/arch__common.dir/isr_tables.c.obj' failed
make[2]: *** [zephyr/arch/common/CMakeFiles/arch__common.dir/isr_tables.c.obj] Error 1
CMakeFiles/Makefile2:2403: recipe for target 'zephyr/arch/common/CMakeFiles/arch__common.dir/all' failed
make[1]: *** [zephyr/arch/common/CMakeFiles/arch__common.dir/all] Error 2
Makefile:103: recipe for target 'all' failed
make: *** [all] Error 2

```

**Expected behavior**
Compilation succeeds.

**Impact**
It is not possible to support an SoC without IRQs implemented.

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain Zephyr SDK
 - Commit SHA or Version used: 2505daf28f378aa968cd4e4a7fbdeaaf3eeeeaa8

--- a/arch/arm/core/aarch32/cortex_a_r/vector_table.S
+++ b/arch/arm/core/aarch32/cortex_a_r/vector_table.S
@@ -23,5 +23,9 @@ SECTION_SUBSEC_FUNC(exc_vector_table,_vector_table_section,_vector_table)
 	ldr pc, =z_arm_prefetch_abort    /* prefetch abort offset  0xc */
 	ldr pc, =z_arm_data_abort        /* data abort     offset 0x10 */
 	nop				 /*                offset 0x14 */
+#ifdef CONFIG_GEN_SW_ISR_TABLE
 	ldr pc, =_isr_wrapper		 /* IRQ            offset 0x18 */
+#else
+	ldr pc, =z_irq_spurious
+#endif
 	ldr pc, =z_arm_nmi               /* FIQ            offset 0x1c */

--- a/arch/arm/core/aarch64/vector_table.S
+++ b/arch/arm/core/aarch64/vector_table.S
@@ -82,7 +82,12 @@ SECTION_SUBSEC_FUNC(exc_vector_table,_vector_table_section,_vector_table)
 
 	/* Current EL with SPx / IRQ */
 	.align 7
-	b	_isr_wrapper
+
+#ifdef CONFIG_GEN_SW_ISR_TABLE
+	b 	_isr_wrapper
+#else
+	b	z_irq_spurious
+#endif
 
 	/* Current EL with SPx / FIQ */
 	.align 7

--- a/tests/kernel/gen_isr_table/src/main.c
+++ b/tests/kernel/gen_isr_table/src/main.c
@@ -326,6 +326,9 @@ void main(void)
 #endif
 #endif /* CONFIG_GEN_SW_ISR_TABLE */
 	rv = TC_PASS;
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-label"
 done:
 	TC_END_RESULT(rv);
 	TC_END_REPORT(rv);

--- a/tests/kernel/gen_isr_table/testcase.yaml
+++ b/tests/kernel/gen_isr_table/testcase.yaml
@@ -9,6 +9,12 @@ tests:
         cc26x2r1_launchxl olimex_stm32_h103
     filter: CONFIG_GEN_ISR_TABLES and CONFIG_ARMV7_M_ARMV8_M_MAINLINE
     tags: interrupt isr_table
+  arch.interrupt.gen_isr_table.disabled:
+    arch_allow: arm
+    extra_configs:
+      - CONFIG_GEN_ISR_TABLES=n
+    tags: interrupt isr_table
+    build_only: true
   arch.interrupt.gen_isr_table.arc:
     arch_allow: arc
     filter: CONFIG_RGF_NUM_BANKS > 1

